packages/providers/src.ts/base-provider.ts

```diff
@@ -218,28 +240,29 @@ export class BaseProvider extends Provider {
     _previousPollingTimestamps: { [key: string]: Timestamp }
 
     _options: ProviderOptions;
-
     readonly anyNetwork: boolean;
     private readonly hederaClient: Client;
     private readonly _mirrorNodeUrl: string; // initial mirror node URL, which is resolved from the provider's network
 
     constructor(network: Networkish | Promise<Network> | HederaNetworkConfigLike, options?: ProviderOptions) {
         logger.checkNew(new.target, Provider);
         super();
-
-        this._options = options || {};
         this._events = [];
         this._emittedEvents = {};
         this._previousPollingTimestamps = {};
         this.formatter = new.target.getFormatter();
         // If network is any, this Provider allows the underlying
         // network to change dynamically, and we auto-detect the
         // current network
+        if(options !== undefined && options !== null){
+            this._options = Object.keys(options).length !== 0 ? options : { headers : {}, retry: DEFAULT_RETRY_OPTIONS };
```

Comment:
Q: Would this work if `options` is passed with only some custom `headers` specified, without a `retry` property?

Reply:
No, but it's fixed now. If you don't specify `retry` property, it will take the default.